src/state/boot.ts

```diff
@@ -31,6 +32,16 @@ function addEventListeners() {
 }
 
 if (globalThis.window) {
-  loadConfig();
+  // loadConfig is now called conditionally from setBitcoinConnectConfig
```

Comment:
I don't think we should make this change, which is quite complicated and I think unnecessary. `loadConfig` will do nothing if no config is saved to local storage.